daemon/double_restart_test.go

```diff
@@ -0,0 +1,137 @@
+package daemon
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/athanorlabs/atomic-swap/coins"
+	"github.com/athanorlabs/atomic-swap/common/types"
+	"github.com/athanorlabs/atomic-swap/monero"
+	"github.com/athanorlabs/atomic-swap/rpcclient"
+	"github.com/athanorlabs/atomic-swap/rpcclient/wsclient"
+	"github.com/athanorlabs/atomic-swap/tests"
+)
+
+// This test starts a swap between Bob and Alice. The nodes restart *twice* after the xmr is locked
+// but before Bob claims. Then, on second restart, Bob should have claimed and
+// Alice should be able to get the XMR.
```

Comment:
The next person maintaining this test will probably be wondering, "why 2?". You had a very specific reason, but I don't think I understood it well enough to describe it in the comment here.

Reply:
yeah I'll add more info to the comment